lang/LangPrimSource/PyrUnixPrim.cpp

```diff
@@ -186,6 +185,79 @@ int prString_POpen(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int prArrayPOpen(struct VMGlobals *g, int numArgsPushed);
+int prArrayPOpen(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrObject *obj;
+
+	PyrSlot *a = g->sp - 1;
+	PyrSlot *b = g->sp;
+	
+#ifdef SC_IPHONE
+	SetInt(a, 0);
+	return errNone;
+#endif
+	
+	if (NotObj(a)) return errWrongType;
+
+	obj = slotRawObject(a);
+	if (!(slotRawInt(&obj->classptr->classFlags) & classHasIndexableInstances))
+		return errNotAnIndexableObject;
+		
+	if( obj->size < 1)
+		return errFailed;
+		
+	PyrSlot filenameSlot;
+	getIndexedSlot(obj, &filenameSlot, 0);
+	if (!isKindOfSlot(&filenameSlot, class_string)) return errWrongType;
+	char filename[slotRawObject(&filenameSlot)->size + 1];
```

Comment:
there is a notion of `PATH_MAX`/`MAX_PATH`. variable length arrays aren't well supported with msvc, which only implements a subset of c99.